### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by VladimirGoncharov**

### CODE REVIEW

**Comment by NikolaevSergey** [comment]
:+1:

File: ios-base/Classes/APIManager.h [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
 + (APIManager*)sharedManager;
 
-- (void)getCurrentDateWithCompleteBlock:(ObjectCallback)block;
+@property (nonatomic, strong, readonly) AFHTTPRequestOperationManager *operationManager;
 
 @end
+
+
+#pragma mark -
+@interface APIManager (Date)
```
Comment (nikitafomin) [inline ios-base/Classes/APIManager.h]: I do not like this way of separating functions to different classes. @NikolaevSergey what do you think?

File: ios-base/Classes/ErrorHandler.m [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        if ([value isKindOfClass:[NSString class]])
+        {
+            if (resultKey)
+            {
+                *resultKey            = key;
+            }
+            
+            if (resultReason)
+            {
+                *resultReason         = value;
```
Comment (nikitafomin) [inline ios-base/Classes/ErrorHandler.m]: This way to coding is confusing for me. To large distance. @NikolaevSergey

Comment (NikolaevSergey) [inline ios-base/Classes/ErrorHandler.m]: @nikitafomin you about lines or space between "resultKey" and "key"?
I don't like so much space between var and value if there is not a lot of values together
But lines is OK.

File: ios-base/Classes/APIManager.h (hunk 1 above)
Comment (NikolaevSergey) [inline ios-base/Classes/APIManager.h]: @nikitafomin I think pragma mark is enough to separate logic in class

Comment (nikitafomin) [inline ios-base/Classes/APIManager.h]: I agree with Sergey

File: ios-base/AppDelegate.m [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
-//  Copyright (c) 2014 FlatStack. All rights reserved.
 //
 
 #import "AppDelegate.h"
 
 @implementation AppDelegate
 
-- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
+- (BOOL)application:(UIApplication *)application
+didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
```
Comment (nikitafomin) [inline ios-base/AppDelegate.m]: @NikolaevSergey what do you think about this style of coding? I about transfer to another line parameters of functions.

Comment (nikitafomin) [inline ios-base/AppDelegate.m]: I don't like how ":" don't match

Comment (NikolaevSergey) [inline ios-base/AppDelegate.m]: It's nice in functions with many arguments, but not when we have 1-2 incoming values

**Comment by nikitafomin** [comment]
You forgot to replace categories with pragma mark as me and @NikolaevSergey discuss here https://github.com/fs/ios-base/pull/43/#discussion_r27732747

**Comment by nikitafomin** [comment]
:+1:

**Pull request merged**